addons/pos_restaurant/static/src/js/PointOfSaleModel.js

```diff
@@ -407,24 +401,58 @@ odoo.define('pos_restaurant.PointOfSaleModel', function (require) {
             this.data.uiState.FloorScreen.isEditMode = false;
             await this.actionShowScreen('FloorScreen');
         },
-        async actionSetTable(table) {
+        /**
+         * Set the given `table` as active, and optionally select the given `orderToSelectId`.
+         * @param {'restaurant.table'} table
+         * @param {'string' | undefined} orderToSelectId
+         */
+        async actionSetTable(table, orderToSelectId) {
             if (this.data.uiState.FloorScreen.isEditMode) {
                 this.data.uiState.FloorScreen.selectedTableId = table.id;
+            } else if (this.exists('pos.order', orderToSelectId)) {
+                await this._setTableWithOrder(table, orderToSelectId);
             } else {
-                this.data.uiState.activeTableId = table.id;
-                await this._syncFromServer(table);
-                if (this.data.uiState.orderIdToTransfer) {
-                    const orderToTransfer = this.getRecord('pos.order', this.data.uiState.orderIdToTransfer);
-                    this.updateRecord('pos.order', orderToTransfer.id, { table_id: table.id });
-                    this.data.uiState.orderIdToTransfer = false;
-                    await this.actionSelectOrder(orderToTransfer);
+                // This is reached when clicking a table from the floorscreen.
+                await this._setTableOnly(table);
+            }
+        },
+        async _setTableWithOrder(table, orderToSelectId) {
+            const orderToSelect = this.getRecord('pos.order', orderToSelectId);
+            if (orderToSelect.table_id !== table.id) {
+                await this.ui.askUser('ErrorPopup', {
+                    title: _t('Set table error'),
+                    body: _t("Can't select order that doesn't belong to the table."),
+                });
+                // TODO jcb: this could be better than showing an error.
```

Comment:
What is your opinion on this? This can happen when there is a programming error, e.g., actionSetTable is called with order that doesn't belong to the table. How do you think should we handle it?
I don't think error is appropriate. I'm thinking we set the table and ignore the given order. Or maybe error is the right one. Perhaps a console error?

Reply:
`throw new Error("Can't select order that doesn't belong to the table.");`

perhaps I should just do that because it's the programmer's fault if that is reached. Any idea?